### DISCUSSION TITLE
Issue with Question Publishing Process

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-inQuiry/Community#117] [Q&A] [answered]
Hi everyone,

I'm encountering an issue with the question publishing process.

Steps I followed:

Question Creation:
I successfully created a questionset and shared it for review. It got success.

Question Review:
I reviewed the question using the  API, and everything worked fine.

Question Publishing:
When I published the question using the API, the response returned a success status (200).

The Issue:
After publishing, when I go to creators side the questioset stills shows in "Review" instead of "Live," which is unexpected.

We have also checked the logs for async-questionset-publish flink job we are getting issue in running the job, Here is the error log for the same 
`Caused by: org.sunbird.job.exception.InvalidInputException: Error while downloading file https://sunbirdsaas.comCONTENT_STORAGE_BASE_PATH/content/assets/do_1141736598468853761425/screenshot-2024-10-29-at-11.53.55-am.png`

Any insights or suggestions would be greatly appreciated!

cc: @pallakartheekreddy @coolbung @baljeetsingh @jona-r @HarishGangula @maheshkumargangula

**Comment by gandham-santhosh** [comment]
@siddhishinde0723 this looks like issue related to the environment variables initialisation related to cloud storage as the application is not constructing the correct path. Can you check the config map associated to this flink job and share here?

**Reply by siddhishinde0723** [reply]
@gandham-santhosh Here is the config map 
#Cloud Storage Config
cloud_storage_type: "azure"
cloud_storage_key: ""
cloud_storage_secret: ""
cloud_storage_endpoint: ""
cloud_storage_container: "content"
master.category.validation.enabled ="Yes"
cloudstorage {
  metadata.replace_absolute_path=true
  metadata.list=["appIcon","posterImage","artifactUrl","downloadUrl","variants","previewUrl","pdfUrl"]
  relative_path_prefix="CONTENT_STORAGE_BASE_PATH"
  read_base_path="https://cdn.sunbirdsaas.com"
  write_base_path=["https://sunbirdsaaspublic.blob.core.windows.net", "https://cdn.sunbirdsaas.com"]
}

**Comment by pallakartheekreddy** [comment, marked as answer]
@siddhishinde0723 As discussed during the call, this issue is occurring because the image upload path through the Question Set Editor is not resolving properly. Please add the URL below to the `cloud_storage_url` array in the portal config, and the issue will be fixed.

```["https://sunbirdsaaspublic.blob.core.windows.net", "https://cdn.sunbirdsaas.com"]```

**Reply by siddhishinde0723** [reply]
@pallakartheekreddy  After updating this now we are able to publish the Question set.
Thank you for your support.
